### ISSUE TITLE
Unit Test 'test_wellmodel' Requires C++14

### ISSUE THREAD

**Issue by bska** [OPM/opm-simulators#1264]
The unit test [`test_wellmodel`](https://github.com/OPM/opm-simulators/blob/master/tests/test_wellmodel.cpp) was introduced in the recent PR #1250.  This unit test features a fixture whose constructor uses `std::make_unique<T>()` and therefore requires C++14.

Was this a conscientious decision and are we now supposed to require C++14 for all of OPM going forward?  If not, a patch that replaces `ptr = make_unique<T>()` with `ptr.reset(new T)` is sufficient to make the unit test compile as C++11.

**Comment by GitPaean** [comment]
Thanks for spotting this.  I specified cmake option file with `-std++0x`, which should be same with `-std++11`. while it turned out in the actual compiling, `-std++14` was added somehow.  I am not sure how this happened.  PR#1265 has been created as requested.

**Comment by akva2** [comment]
ref https://github.com/OPM/opm-common/commit/4d31c7efdc2314dc384ccae4c3ccc6c584f5262d

**Comment by GitPaean** [comment]
> ref OPM/opm-common@4d31c7e

Does this mean OPM has already accepted `C++14` standard?

**Comment by akva2** [comment]
no, dune 2.5 is not mandatory. however c++14 is mandatory for dune 2.5.

**Comment by bska** [comment]
> ref OPM/opm-common@4d31c7e

Okay, that explains why I caught this in my build test today.  In this particular build I happened to be using GCC 4.9 whose (incomplete) C++14 support is activated by the flag `-std=c++1y` (not amongst the alternatives listed in the above commit).  In a different build test, on another machine, I used GCC 5.4 which supports the `-std=c++14` flag and this issue went unnoticed.
